src/aeso_ast_infer_types.erl

```diff
@@ -1024,6 +1104,8 @@ check_typedef_sccs(Env, TypeMap, [{acyclic, Name} | SCCs], Acc) ->
                     Env2 = check_fields(Env1, TypeMap, RecTy, Fields),
                     check_typedef_sccs(Env2, TypeMap, SCCs, Acc1);
                 {variant_t, Cons} ->
+                    ets_insert(type_vars_variance, {Env#env.namespace ++ qname(D),
```

Comment:
What does it mean here?

Reply:
I probably don't get the question.

Here we have access to the datatype's type variables `Xs` and the constructors for the datatype `Cons`, and I use that to infer the type variance for these type variables.

Are you asking about something else?

Reply:
...and now I am happy that we don't have recursive datatypes